src/Microsoft.DocAsCode.App/DocfxProject.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.DocAsCode.Common;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DocAsCode
+{
+    /// <summary>
+    /// Provides options to be used with <see cref="DocfxProject.Build(BuildOptions)"/>.
+    /// </summary>
+    public class BuildOptions
+    {
+        /// <summary>
+        /// Gets a read-only, singleton instance of <see cref="BuildOptions"/> that uses the default configuration.
+        /// </summary>
+        public static BuildOptions Default { get; } = new();
+
+        /// <summary>
+        /// Get or sets a value that indicates whether to export model as JSON as part of the build.
+        /// </summary>
+        public bool Json { get; init; }
```

Comment:
naming: ExportJsonModel?

question: is this for metadata or build step or both?

Reply:
It is for build step, do we export JSON model for metadata step?

I went back and forward between `Json` and `ExportJson`, and tentatively named it `json` which looks similar to `typedoc`'s [`json`](https://typedoc.org/guides/options/#json) option:

I do want the name to be the same as command line args similar to `typedoc`. `--exportRawModel` and `--exportViewModel` are the args we have today, I believe `--exportViewModel` is something we want to get rid of with a newer template system because it is tied to the intermediate javascript execution step.

Even as I'm typing this I realize there is a semantic difference in "how `--exportRawModel` behaves today" vs "how I want it to behave". Today it export JSON in addition to HTML, but I would like to only export JSON files under this mode. So I might remove this option for now until we've settled down the semantics. It is used for testing purposes anyway.